File: app/src/main/java/digispark/tech/ganapatiaarti/firebase/NotificationHelper.java
package digispark.tech.ganapatiaarti.firebase;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.media.RingtoneManager;
import android.net.Uri;

import androidx.core.app.NotificationCompat;

import digispark.tech.ganapatiaarti.R;
import digispark.tech.ganapatiaarti.constants.Constant;

public class NotificationHelper {
    private static final String CHANNEL_ID = "channel_id_01";
    private static final String NAME = "Ganapati_noti";
    private static final String DESCRIPTION = "Ganapati aarti description";

    public static void createNotificationChannel(Context context) {
        NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
        if (notificationManager != null){
            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
                int importance = NotificationManager.IMPORTANCE_HIGH;
                NotificationChannel mChannel = new NotificationChannel(CHANNEL_ID, NAME, importance);
                mChannel.setDescription(DESCRIPTION);
                mChannel.enableLights(true);
                mChannel.setLightColor(Color.RED);
                mChannel.enableVibration(true);
                mChannel.setLockscreenVisibility(Notification.VISIBILITY_PRIVATE);
                mChannel.setVibrationPattern(new long[]{100, 200, 300, 400, 500, 400, 300, 200, 400});
                mChannel.setShowBadge(true);
                notificationManager.createNotificationChannel(mChannel);
            }
        }
    }


    public static void showNotification(Context context, String notiTitle, String notiText){
        try {
            NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
            if (notificationManager != null){
                PendingIntent p = getPendindIntent(context, notiTitle);
                Uri alarmSound= RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
                NotificationCompat.Builder builder = new NotificationCompat.Builder(context, CHANNEL_ID);
                builder.setSmallIcon(R.mipmap.ic_launcher);
                builder.setLargeIcon(BitmapFactory.decodeResource(context.getResources(),
                        R.mipmap.ic_launcher));
                builder.setContentTitle(Constant.APP_NAME);
                builder.setDefaults(Notification.DEFAULT_ALL);
                builder.setPriority(Notification.PRIORITY_HIGH);
                builder.setPriority(NotificationCompat.PRIORITY_HIGH);
                builder.setContentTitle(notiTitle);
                builder.setContentText(notiText);
                builder.setSound(alarmSound);
                builder.setAutoCancel(true);
                builder.setVibrate(new long[]{0});
                if (p != null)
                    builder.setContentIntent(p);
                notificationManager.notify(Constant.NOTIFY_ID, builder.build());
            }
        }
        catch (Exception e) {
            e.printStackTrace();
        }
    }

    private static PendingIntent getPendindIntent(Context context, String notiTitle){
        if (notiTitle.contains(Constant.UPDATE)){
            Uri uri = Uri.parse(Constant.PLAY_STORE_LINK);
            Intent updateAppIntent = new Intent();
            updateAppIntent.setAction(Intent.ACTION_VIEW);
            // After pressing back button from google play will continue to app
            updateAppIntent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY | Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET |
                    Intent.FLAG_ACTIVITY_MULTIPLE_TASK);
            updateAppIntent.setData(uri);
            return PendingIntent.getActivity(context, Constant.PENDING_INTENT_REQUEST_CODE, updateAppIntent, PendingIntent.FLAG_UPDATE_CURRENT);
        }
        return null;
    }

    public static void clearNotifications(Context context) {
        if (context != null){
            NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
            if (notificationManager!= null)
                notificationManager.cancelAll();
        }
    }
}

File: app/src/main/java/digispark/tech/ganapatiaarti/firebase/NotificationHelper.kt
package digispark.tech.ganapatiaarti.firebase

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.graphics.BitmapFactory
import android.graphics.Color
import android.media.RingtoneManager
import android.net.Uri

import androidx.core.app.NotificationCompat

import digispark.tech.ganapatiaarti.R
import digispark.tech.ganapatiaarti.constants.Constant

object NotificationHelper {
    private val CHANNEL_ID = "channel_id_01"
    private val NAME = "Ganapati_noti"
    private val DESCRIPTION = "Ganapati aarti description"

    fun createNotificationChannel(context: Context) {
        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        if (notificationManager != null) {
            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
                val importance = NotificationManager.IMPORTANCE_HIGH
                val mChannel = NotificationChannel(CHANNEL_ID, NAME, importance)
                mChannel.description = DESCRIPTION
                mChannel.enableLights(true)
                mChannel.lightColor = Color.RED
                mChannel.enableVibration(true)
                mChannel.lockscreenVisibility = Notification.VISIBILITY_PRIVATE
                mChannel.vibrationPattern = longArrayOf(100, 200, 300, 400, 500, 400, 300, 200, 400)
                mChannel.setShowBadge(true)
                notificationManager.createNotificationChannel(mChannel)
            }
        }
    }


    fun showNotification(context: Context, notiTitle: String, notiText: String) {
        try {
            val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            if (notificationManager != null) {
                val p = getPendindIntent(context, notiTitle)
                val alarmSound = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
                val builder = NotificationCompat.Builder(context, CHANNEL_ID)
                builder.setSmallIcon(R.mipmap.ic_launcher)
                builder.setLargeIcon(BitmapFactory.decodeResource(context.resources,
                        R.mipmap.ic_launcher))
                builder.setContentTitle(Constant.APP_NAME)
                builder.setDefaults(Notification.DEFAULT_ALL)
                builder.priority = Notification.PRIORITY_HIGH
                builder.priority = NotificationCompat.PRIORITY_HIGH
                builder.setContentTitle(notiTitle)
                builder.setContentText(notiText)
                builder.setSound(alarmSound)
                builder.setAutoCancel(true)
                builder.setVibrate(longArrayOf(0))
                if (p != null)
                    builder.setContentIntent(p)
                notificationManager.notify(Constant.NOTIFY_ID, builder.build())
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    private fun getPendindIntent(context: Context, notiTitle: String): PendingIntent? {
        if (notiTitle.contains(Constant.UPDATE)) {
            val uri = Uri.parse(Constant.PLAY_STORE_LINK)
            val updateAppIntent = Intent()
            updateAppIntent.action = Intent.ACTION_VIEW
            // After pressing back button from google play will continue to app
            updateAppIntent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY or Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET or
                    Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
            updateAppIntent.data = uri
            return PendingIntent.getActivity(context, Constant.PENDING_INTENT_REQUEST_CODE, updateAppIntent, PendingIntent.FLAG_UPDATE_CURRENT)
        }
        return null
    }

    fun clearNotifications(context: Context?) {
        if (context != null) {
            val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            notificationManager?.cancelAll()
        }
    }
}
